pkg/api/controller.go

```diff
@@ -94,6 +95,91 @@ type Controller struct {
 	PathProvider          upload.PathProvider
 }
 
+func decodeGCUncommittedMark(mark string) (*catalog.GCUncommittedMark, error) {
+	data, err := base64.StdEncoding.DecodeString(mark)
+	if err != nil {
+		return nil, err
+	}
+	var result catalog.GCUncommittedMark
+	err = json.Unmarshal(data, &result)
+	if err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+func encodeGCUncommittedMark(mark *catalog.GCUncommittedMark) (string, error) {
+	var buf bytes.Buffer
+	encoder := base64.NewEncoder(base64.StdEncoding, &buf)
+	err := json.NewEncoder(encoder).Encode(mark)
+	if err != nil {
+		return "", err
+	}
+	return buf.String(), nil
+}
+
+func (c *Controller) PrepareGarbageCollectionUncommitted(w http.ResponseWriter, r *http.Request, body PrepareGarbageCollectionUncommittedJSONRequestBody, repository string) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.PrepareGarbageCollectionUncommittedAction,
+			Resource: permissions.RepoArn(repository),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "prepare_garbage_collection_uncommitted", r, repository, "", "")
+
+	// TODO(barak): move this code the a Decode func in catalog
+	var mark *catalog.GCUncommittedMark
+	continuationToken := StringValue(body.ContinuationToken)
+	if continuationToken != "" {
+		var err error
+		mark, err = decodeGCUncommittedMark(continuationToken)
+		if err != nil {
+			c.Logger.WithError(err).
+				WithFields(logging.Fields{"repository": repository, "continuation_token": continuationToken}).
+				Error("Failed to decode gc uncommitted continuation token")
+			writeError(w, http.StatusBadRequest, "invalid continuation token")
+			return
+		}
+	}
+
+	uncommittedInfo, err := c.Catalog.PrepareGCUncommitted(ctx, repository, mark)
+	if err != nil {
+		c.Logger.WithError(err).
+			WithFields(logging.Fields{"repository": repository, "continuation_token": continuationToken}).
+			Error("PrepareGCUncommitted failed")
+		c.handleAPIError(ctx, w, err)
+		return
+	}
+
+	// continuation token
+	var nextContinuationToken *string
+	if uncommittedInfo.Mark != nil {
+		nextMark, err := encodeGCUncommittedMark(uncommittedInfo.Mark)
```

Comment:
Should remove the encode method from mark

Reply:
Don't think there is Encode anymore